docs/understanding-radicle/how-radicle-works.md

```diff
@@ -0,0 +1,334 @@
+
+### Seed nodes
+
+Seed nodes are the data availability layer for the Radicle network, designed to ensure that even if everyone in your
+peer-to-peer network is offline, you can always retrieve Radicle project data from a seed node. Seed nodes store your
+identity, but not your private keys, which means they can't take action on your behalf.
+
+Seed nodes are based on fully [open-source code](https://github.com/radicle-dev/radicle-client-services), and Radicle
+currently maintains three seed nodes &mdash;
+[willow.radicle.garden](https://app.radicle.xyz/seeds/willow.radicle.garden),
+[pine.radicle.garden](https://app.radicle.xyz/seeds/pine.radicle.garden), and
+[maple.radicle.garden](https://app.radicle.xyz/seeds/maple.radicle.garden) &mdash; with identical functionality.
+
+Like each person using Radicle, seed nodes store identities, project metadata, and project data using Git, in a format
+that looks a like this:
+
+```jsx
+refs/
+  namespaces/
+    <project-one>/
+      refs/
+        heads/
+        remotes/
+          alice/
+          eve/
+    <project-two>/
+      refs/
+        heads/
+        remotes/
+          bob/
+          tom/
+          alice/
+          eve/
+    ...
+```
+
+Seed nodes have two primary components.
+
+- The **HTTP API** (`radicle-http-api`), which allows for browsing Radicle projects on the web. It's the backend of the
+  [Radicle web app](https://app.radicle.xyz/), but the API is entirely [open
+  source](https://github.com/radicle-dev/radicle-interface) for anyone else to develop additional clients.
+- The **Git server** (`radicle-git-server`), which acts as a bridge between the seed node's Git server and those who use
+  it, replicating Git data and Radicle identities.
+
+The Git server is the primary mode of replication at the moment, and it behaves in a push-pull fashion, just like
+running `git push` or `git pull` to a repository you have with GitHub or any other Git provider. A lot happens behind
+the scenes, like pushing identities, validating signatures, and replicating additional data like peers and sub-trees.
+
+We could stop here, because seed nodes are already decentralized and open source. If you want to run a private Radicle
+network, you can [host your own seed node](https://github.com/radicle-dev/radicle-client-services) for true peer-to-peer
+collaboration.
+
+But, the push-pull relationship between a local machine, seed node, and collaborators means that the project's
+location—which seed node contains the most up-to-date state—is a prominent part of the user experience. You *have* to
+know and remember which seed node you're working with.
+
+Radicle's peer-to-peer protocol will solve this problem.
+
+:::info
+
+The open-source code behind Radicle's seed nodes is maintained by the clients team at
+[`radicle-client-services`](https://github.com/radicle-dev/radicle-client-services).
+
+:::
+
+## Peer-to-peer protocol
```

Comment:
My only note here is that people who are familiar with the idea of p2p will expect devices to be able to gossip directly with each other.

Unfortunately, this problem wasn't solved in today's current state because ISPs and firewalls are a pain in the ass.

So while gossip does happen, edge nodes will only be able to directly replicate from seeds and not other edges.

Reply:
Can you dig into edge nodes a bit more? I haven't seen that term come up in the context of Radicle, so I'm curious about the implementation.

Reply:
Ya sorry, edge nodes are generally ephemeral devices on the _edge_ of the network, i.e. laptops, phones, etc. Opposed to servers which are the always-on, inner nodes in the network.

To put my original comment another way, "our laptops can't directly connect to each other because of pesky ISP routers"